app/forms/profile_form.rb

```diff
@@ -0,0 +1,54 @@
+class ProfileForm
+  include ActiveModel::Model
+  include Virtus.model
+
+  AFFILIATION_VALUES = [ 'undergraduate', 'graduate', 'technical_college', 'college', 'working', 'the_others' ]
+
+  validates :affiliation, inclusion: { in: AFFILIATION_VALUES }
+  validates :school,     length: { maximum: 50 }
+  validates :faculty,    length: { maximum: 50 }
+  validates :department, length: { maximum: 50 }
+  validates :laboratory, length: { maximum: 50 }
+  validates :content,    length: { maximum: 200 }
+  validate :at_least_one_parameter
+
+  attribute :affiliation, String
+  attribute :school,      String
+  attribute :faculty,     String
+  attribute :department,  String
+  attribute :laboratory,  String
+  attribute :content,     String
+  attribute :user_id,     Integer
+
+  attr_accessor :user, :profile
+
+  def assign_attributes(params = {}, user)
```

Comment:
記事の通りに作成した方がいいかと。assign_attributesはRailsのメソッドをオーバーライドしてるので、引数の数は変えてはいけないかと思います。
```suggestion
  def assign_attributes(params = {})
```